src/Tmds.Ssh/SshAgent.cs

```diff
@@ -23,60 +23,77 @@ sealed class SshAgent : IDisposable
         private const uint SSH_AGENT_RSA_SHA2_512 = 4;
         private const int MaxPacketSize = int.MaxValue; // We trust the SSH Agent.
 
-        private static EndPoint? _defaultEndPoint;
-
         public struct Identity
         {
             public string Comment { init; get; }
             public byte[] PublicKey { init; get; }
         }
 
-        public static EndPoint? DefaultEndPoint
+        public static string? DefaultEndPoint
         {
             get
             {
-                string? sshAuthSock = Environment.GetEnvironmentVariable("SSH_AUTH_SOCK");
-
-                if (OperatingSystem.IsWindows())
-                {
-                    return null;
-                }
-
-                if (string.IsNullOrEmpty(sshAuthSock))
-                {
-                    return null;
-                }
-
-                _defaultEndPoint ??= new UnixDomainSocketEndPoint(sshAuthSock);
-
-                return _defaultEndPoint;
+                return OperatingSystem.IsWindows()
+                    ? "openssh-ssh-agent"
+                    : Environment.GetEnvironmentVariable("SSH_AUTH_SOCK");
             }
         }
 
-        private readonly EndPoint _endPoint;
+        private readonly string _endPoint;
         private readonly SequencePool _sequencePool;
 
-        private SocketSshConnection? _agentConnection;
+        private StreamSshConnection? _agentConnection;
 
-        public SshAgent(EndPoint endPoint, SequencePool sequencePool)
+        public SshAgent(string endPoint, SequencePool sequencePool)
         {
             _endPoint = endPoint;
             _sequencePool = sequencePool;
         }
 
         public async ValueTask<bool> TryConnect(CancellationToken cancellationToken)
         {
+            NamedPipeClientStream? pipe = null;
```

Comment:
After https://github.com/tmds/Tmds.Ssh/pull/291/files#r1895384096, try changing this to make the two branches produce a `Stream` (and move these declarations local to their branch), and then create the `StreamSshConnection` after the branches.

Reply:
Done, let me know if you preferred what I've done in a different way.